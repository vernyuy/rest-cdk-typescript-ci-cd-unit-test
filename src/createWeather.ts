// 'use strict';

import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
import * as AWS from "aws-sdk";

const docClient = new AWS.DynamoDB.DocumentClient();
const tableName = process.env.TABLE_NAME as string;
// const region = process.env.Region;

export const lambdaHandler = async (
  event: any
): Promise<APIGatewayProxyResult> => {
  let response: APIGatewayProxyResult;
  const id = Math.floor(Math.random() * 1000).toString();
  const body = JSON.parse(event.body);
  const weatherItem = {
    id: id,
    weather: body.weather as string,
    town: body.town as string,
  };
  try {
    const test = await docClient
      .put({
        TableName: tableName,
        ReturnConsumedCapacity: "TOTAL",
        Item: weatherItem,
      })
      .promise();
      console.log(test);
      response = {
        statusCode: 200,
        body: JSON.stringify({
          message: "Weather item created Successfully",
        }),
      };
  } catch (err: unknown) {
    console.log(err);
    response = {
      statusCode: 500,
      body: JSON.stringify({
        message:
          err instanceof Error
            ? err.message
            : "An error occured while creating weather.",
      }),
    };
  }
  return response;
};
