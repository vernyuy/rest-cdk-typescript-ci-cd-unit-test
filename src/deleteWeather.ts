import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
import * as AWS from "aws-sdk";

const tableName = process.env.TABLE_NAME as string;
const region = process.env.Region;
const client = new AWS.DynamoDB.DocumentClient();


export const lambdaHandler = async  (
  event: any
): Promise<APIGatewayProxyResult> => {
  let response: APIGatewayProxyResult;
  const weatherId = event.pathParameters.id as string;
  try {
    const res = await client
      .delete({
        TableName: tableName,
        Key: {
          id: weatherId,
        },
      })
      .promise();
    response = {
      statusCode: 200,
      body: JSON.stringify({
        message: "Weather Item deleted successfully",
      }),
    };
  } catch (err: unknown) {
    console.log(err);
    response = {
      statusCode: 500,
      body: JSON.stringify({
        message: err instanceof Error ? err.message : "some error happened",
      }),
    };
  }
  return response;
};
