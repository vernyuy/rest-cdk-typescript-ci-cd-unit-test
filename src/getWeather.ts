import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
import * as AWS from "aws-sdk";

const docClient = new AWS.DynamoDB.DocumentClient();
const tableName = process.env.TABLE_NAME as string;
const region = process.env.Region;
// const client = new DynamoDBClient({ region: region });

export const lambdaHandler = async (
  event: any
): Promise<APIGatewayProxyResult> => {
  let response: APIGatewayProxyResult;
  console.log(event);
  console.log(event.pathParameters.id);
  const weatherId = event.pathParameters.id;
  var params = {
    Key: {
      id: {
        S: weatherId as string,
      },
    },
    TableName: process.env.TABLE_NAME,
  };
  try {
    const res = await docClient
      .get({
        Key: {
          id: {
            S: weatherId as string,
          },
        },
        TableName: tableName,
      })
      .promise();
    response = {
      statusCode: 200,
      body: JSON.stringify({
        weather: res.Item,
      }),
    };
  } catch (err: unknown) {
    console.log(err);
    response = {
      statusCode: 500,
      body: JSON.stringify({
        message: err instanceof Error ? err.message : "some error happened",
      }),
    };
  }
  return response;
};
