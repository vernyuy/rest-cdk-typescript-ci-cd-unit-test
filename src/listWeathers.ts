import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
import * as AWS from "aws-sdk";

const region = process.env.Region;
const docClient = new AWS.DynamoDB.DocumentClient();
const tableName = process.env.TABLE_NAME as string;

export const lambdaHandler = async (): Promise<APIGatewayProxyResult> => {
  let response: APIGatewayProxyResult;
  try {
    const res = await docClient.scan({ TableName: tableName }).promise();
    response = {
      statusCode: 200,
      body: JSON.stringify(res.Items),
    };
  } catch (err: unknown) {
    console.log(err);
    response = {
      statusCode: 500,
      body: JSON.stringify({
        message: err instanceof Error ? err.message : "some error happened",
      }),
    };
  }
  return response;
};
