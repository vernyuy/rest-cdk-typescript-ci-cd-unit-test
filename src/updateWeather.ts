import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
import * as AWS from "aws-sdk";

const tableName = process.env.TABLE_NAME as string;
const region = process.env.Region;
const client = new AWS.DynamoDB.DocumentClient();

export const lambdaHandler = async (
  event: any
): Promise<APIGatewayProxyResult> => {
  let response: APIGatewayProxyResult;
  const weather_id = event.pathParameters.id;
  const body = JSON.parse(event.body);
  const weather = body.weather;
  const town = body.town as string;
  const params = {
    Key: {
      id: weather_id as string,
    },
    UpdateExpression: "set weather = :weather, town = :town",
    ExpressionAttributeValues: {
      ":weather": {
        S: weather as string,
      },
      ":town": {
        S: town as string,
      },
    },
    returnValues: "UPDATED_NEW",
    TableName: tableName,
  };
  try {
    const res = await client.update(params).promise();
    response = {
      statusCode: 200,
      body: JSON.stringify({
        message: "Weather Successfully updated",
      }),
    };
  } catch (err: unknown) {
    console.log(err);
    response = {
      statusCode: 500,
      body: JSON.stringify({
        message: err instanceof Error ? err.message : "some error happened",
      }),
    };
  }
  return response;
};
