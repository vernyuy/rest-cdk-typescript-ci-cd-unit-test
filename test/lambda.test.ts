import { lambdaHandler as createWeather } from "../src/createWeather";
import { lambdaHandler as getWeather } from "../src/getWeather";
import { lambdaHandler as deleteWeather } from "../src/deleteWeather";
import { lambdaHandler as updateWeather } from "../src/updateWeather";
import { lambdaHandler as listWeathers } from "../src/listWeathers";

test("create weather", async () => {
  const result = await createWeather({
    body: JSON.stringify({
      weather: "hot",
      town: "Douala",
    }),
  });
  expect(result).toMatchObject({
    body: expect.any(String),
    statusCode: expect.any(Number),
  });
});

test("get weather", async () => {
  const result = await getWeather({
    pathParameters: {
      id: 1,
    },
  });
  expect(result).toMatchObject({
    body: expect.any(String),
    statusCode: expect.any(Number),
  });
});

test("delete weather", async () => {
  const result = await deleteWeather({
    pathParameters: {
      id: 1,
    },
  });
  expect(result).toMatchObject({
    body: expect.any(String),
    statusCode: expect.any(Number),
  });
});

test("update weather", async () => {
  const result = await updateWeather({
    body: JSON.stringify({
      weather: "hot",
      town: "Douala",
    }),
    pathParameters: {
      id: 1,
    },
  });
  expect(result).toMatchObject({
    body: expect.any(String),
    statusCode: expect.any(Number),
  });
});

test("list weathers", async () => {
  const result = await listWeathers();
  expect(result).toMatchObject({
    body: expect.any(String),
    statusCode: expect.any(Number),
  });
});
