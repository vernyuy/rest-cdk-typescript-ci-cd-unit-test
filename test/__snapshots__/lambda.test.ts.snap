// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create weather 1`] = `
{
  "body": "{"message":"Missing required key 'TableName' in params"}",
  "statusCode": 500,
}
`;
